### PR TITLE
allowing Queryable to be used with multiple table names

### PR DESCRIPTION
**Pull request by snf**
Related to https://github.com/diesel-rs/diesel/discussions/3779

Couldn't get `cargo test --features "postgres"` to work in my machine but opening the PR. Getting back to this later today

### CODE REVIEW

File: diesel_derives/tests/queryable.rs [hunk 1]
```diff
@@ -39,3 +39,24 @@ fn tuple_struct_without_column_name_annotations() {
...
+    #[derive(Debug, Clone, PartialEq, Eq, Queryable)]
+    #[diesel(table_name = users)]
+    #[diesel(table_name = users_)]
+    struct NewUser {
+        name: String,
+        hair_color: String,
+    }
+
+    let conn = &mut connection();
+    let data = select(sql::<(String, String)>("'red', 'red'")).get_result(conn);
```
Comment (weiznich) [inline diesel_derives/tests/queryable.rs:54]:
```suggestion
    let data = select(sql::<(diesel::sql_types::Text, [String](diesel::sql_types::Text))>("'red', 'red'")).get_result(conn);
```

**Comment by snf** [comment]
Should be good now. Sorry about the spam, didn't spend much time checking why the tests wouldn't compile locally

**Review by weiznich — APPROVED** [review]
Looks good.

No worries about needing some tries to get everything in place. The CI explicitly there for testing, so its fine to use it.

**Comment by 1Dragoon** [comment]
I'm confused about this one, master branch has this change intact while listing diesel_derives on 2.1.0, yet diesel from crates.io version 2.1.4 is referencing derives 2.1.1, which doesn't have this change.

Am I missing something?

**Comment by weiznich** [comment]
We just did not increase the derive version on the master branch after issuing patch releases from the 2.1.x branch. Changes land usually on the master branch. If necessary they are sometimes backported to the latest release branch. Releases are then issued from there.

**Comment by 1Dragoon** [comment]
Any good reason not to have it backported?

**Comment by weiznich** [comment]
Backport do need someone that drives the backport forward and prepares a new release. That hasn't happen for this PR. 

If you are interested in a patch release that includes this fix feel free to prepare a PR that backports this (and the other PR's marked as maybe-backport) and that contains a draft for the release notes. If that happens I can cut a release afterwards.

**Comment by 1Dragoon** [comment]
I tried to look through some other backport PRs to see how it's done but I'm not sure if I understand it right. Is there any kind of documentation for how you do it? I use git all the time but my knowledge on it only goes an inch deep -- just enough to do what i need to do.

**Comment by weiznich** [comment]
https://github.com/diesel-rs/diesel/pull/3803 is an example for such backport PR. Essentially these are created by:

1. Checking out the target branch (git checkout 2.1.x)
2. Creating a new backport branch based on the target branch (git checkout -b backports/2.1.4)
3. Cherry-picking the releveant merge commits in chronological order to the (git cherry-pick commit-hash)
4. Bumping the versions in the relevant crate
5. Filling in a section in `Changelog.md` to describe which fixes are in this PR
6. Opening the PR

**Pull request merged**